scripts/experiment_0_generate_simulated_data/nbconverted/4_similarity_analysis.py

```diff
@@ -0,0 +1,378 @@
+    
+    # SVCCA
+    svcca_results = cca_core.get_cca_similarity(original_data_df.T,
+                                          batch_data_df.T,
+                                          verbose=False)
+    
+    output_list.append(np.mean(svcca_results["cca_coef1"]))
+
+# Convert output to pandas dataframe
+svcca_raw_df = pd.DataFrame(output_list, columns=["svcca_mean_similarity"], index=num_batches)
+svcca_raw_df
+
+
+# In[7]:
+
+
+# Plot
+svcca_raw_df.plot()
+
+
+# ## Check positive control
+# We want to verify that SVCCA(input, input) = 1.  This does not seem to be the case, but I'm not sure why
+
+# In[8]:
+
+
+# Check datasets batch 1 and original are the same
+batch_1_file = os.path.join(
+    batch_dir,
+    "Batch_1.txt")
+print(batch_1_file)
+
+batch_1 = pd.read_table(
+    batch_1_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+    
+batch_1.head(10)
+
+
+# In[9]:
+
+
+i = 1
+batch_other_file = os.path.join(
+    batch_dir,
+    "Batch_"+str(i)+".txt")
+print(batch_other_file)
+
+batch_other = pd.read_table(
+    batch_other_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+batch_other.head(10)
+
+
+# In[10]:
+
+
+# Calculate SVCCA(batch_1, batch_1)
+svcca_results_batch1_itself = cca_core.get_cca_similarity(batch_1.T,
+                                          batch_1.T,
+                                          verbose=False)
+np.mean(svcca_results_batch1_itself["cca_coef1"])
+
+
+# In[11]:
+
+
+# Caluclate SVCCA(batch_1, batch_1 variable)
+svcca_results_batch1_other = cca_core.get_cca_similarity(batch_1.T,
+                                          batch_other.T,
+                                          verbose=False)
+np.mean(svcca_results_batch1_other["cca_coef1"])
+
+
+# ## Calculate Similarity using PCA projection of batched data
+
+# In[24]:
+
+
+output_list = []
+
+for i in num_batches:
+    print('Calculating SVCCA score for 1 batch vs {} batches..'.format(i))
+    
+    # Get batch 1
+    batch_1_file = os.path.join(
+        batch_dir,
+        "Batch_1.txt")
+
+    batch_1 = pd.read_table(
+        batch_1_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+
+    # PCA projection
+    pca = PCA(n_components=num_PCs)
+
+    # Use trained model to encode expression data into SAME latent space
+    original_data_PCAencoded = pca.fit_transform(batch_1)
+
+
+    original_data_PCAencoded_df = pd.DataFrame(original_data_PCAencoded,
+                                         index=batch_1.index
+                                         )
+    
+    # All batches
+    batch_other_file = os.path.join(
+        batch_dir,
+        "Batch_"+str(i)+".txt")
+
+    batch_other = pd.read_table(
+        batch_other_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+    
+    print("Using batch {}".format(i))
+    
+    # Use trained model to encode expression data into SAME latent space
+    batch_data_PCAencoded = pca.fit_transform(batch_other)
```

Comment:
As discussed, based on my understanding of how SVCCA works (SVD and then CCA to map subspaces to a shared space that maximizes the correlation between the two subspaces), I don't think you are required to have the two input spaces be in the same space initially because SVCCA should do this.